CRM/Membershipextrasimporterapi/EntityImporter/RecurContribution.php

```diff
@@ -0,0 +1,230 @@
+<?php
+
+/**
+ * Imports the recurring contribution record.
+ *
+ * Class CRM_Membershipextrasimporterapi_EntityImporter_RecurContribution
+ */
+class CRM_Membershipextrasimporterapi_EntityImporter_RecurContribution {
+
+  private $rowData;
+
+  private $contactId;
+
+  public function __construct($rowData, $contactId) {
+    $this->rowData = $rowData;
+    $this->contactId = $contactId;
+  }
+
+  public function import() {
+    $recurContributionId = $this->getRecurContributionIdIfExist();
+    if ($recurContributionId) {
+      return $recurContributionId;
+    }
+
+    $sqlParams = $this->prepareSqlParams();
+    $sqlQuery = "INSERT INTO `civicrm_contribution_recur` (`contact_id` , `amount` , `currency` , `frequency_unit` , `frequency_interval` , `installments` ,
+            `start_date`, `contribution_status_id`, `payment_processor_id` , `financial_type_id` , `payment_instrument_id`, `auto_renew`, `create_date`,
+            `next_sched_contribution_date`, `cycle_day`) 
+            VALUES (%1, %2, %3, %4, %5, %6, %7, %8, %9 ,%10, %11, %12, %13, %14, %15)";
+    CRM_Core_DAO::executeQuery($sqlQuery, $sqlParams);
+
+    $dao = CRM_Core_DAO::executeQuery('SELECT LAST_INSERT_ID() as recur_contribution_id');
+    $dao->fetch();
+    $recurContributionId = $dao->recur_contribution_id;
+
+    $sqlQuery = "INSERT INTO `civicrm_value_contribution_recur_ext_id` (`entity_id` , `external_id`) 
```

Comment:
Membershipextras extension is a dependency for this extension, and the whole purpose of it is to import Payment plans in a way that is compatible with Membershipextras, and since the performance is vital here, I want to ignore any unnecessary extra SQL query, so I think I will keep it like that here.